usl_pipeline/usl_lib/usl_lib/shared/geo_data.py

```diff
@@ -42,3 +42,41 @@ class Elevation:
 """The bounding box represented by a tuple (min-x, min-y, max-x, max-y).
 """
 BoundingBox = Tuple[float, float, float, float]
+
+
+def bounding_box_intersection(bbox1: BoundingBox, bbox2: BoundingBox) -> bool:
```

Comment:
might also consider something like

```python
@dataclass(slots=True)
class BoundingBox:
    min_x: float
    min_y: float
    max_x: float
    max_y: float

    def intersects(self, other: "BoundingBox") -> bool:

    def contains(self, other: "BoundingBox") -> bool:
```

Reply:
This is a good idea but I had to refactor a lot of code. :)
Also it might be less efficient now in polygon chunker to go over 1m polygons and do binning into chunks. Will see if some further optimization will be needed.